markdown/solace-js-mqtt-pg/solace-js-mqtt-pg.md

```diff
@@ -0,0 +1,349 @@
+
+## Configure Prisma Client JS on the Consumer to Create Items in the Database
+
+Duration: 0:10:00
+
+[Prisma Client JS](https://github.com/prisma/prisma-client-js) is an auto-generated database client that enables type-safe database access and reduces boilerplate. You can use it as an alternative to traditional ORMs such as Sequelize, TypeORM or Knex.js. Prisma Client JS is part of the “Prisma Framework,” which is a set of open source tools to access, migrate, and administer a variety of databases (Postgres, MySQL, MongoDb). There’s been a lot of buzz around it, especially in the GraphQL community, so I wanted to give it a try. After playing around with it, in my opinion it lives up to the hype.
+
+I’m going to speed through the steps required to get Prisma up and running, but I recommend that you read Prisma’s GitHub README top to bottom for a better appreciation of what the library offers.
+
+**🏃Steps to get started:**
+
+First, download Prisma’s CLI using this command:
+
+```
+npm install -g prisma2
+```
+
+Then, make sure you’re in the mqtt-consumer directory, and run:
+
+```
+prisma2 init
+```
+
+You should see the following output in your terminal:
+
+![Prisma2 init](img/prisma2-init.png)
+
+At this point, navigate to `mqtt-consumer/prisma/.env` and fill in `DATABASE_URL` with the value you find on your Heroku dashboard:
+
+![Prisma2 database uri](img/prisma2-database-uri.png)
+
+With this environment variable filled out, you’re ready to see why people love Prisma. From the mqtt-consumer directory, run the following commands:
+
+```
+prisma2 introspect
+prisma2 generate
+```
+
+You should see the following output in your terminal:
+
+![Prisma2 generate](img/prisma2-generate-1.png)
+
+… and if you look at your schema.prisma now, you’ll see that it has inserted a new code block:
+
+![Prisma2 generate](img/prisma2-generate-2.png)
+
+That’s it for setup. When you ran the previous two commands, Prisma2 automatically generated data models based on your database’s existing tables and installed a custom Prisma2 client package inside your node_modules that includes all the methods related to CRUD operations on your objects. Don’t run `npm i @prisma/client`, the generate step does this for you. This easy-to-use client plus being able to store declarative data models in version control (schema.prisma file) is why people love Prisma.
+
+Now all that’s left is to uncomment the client code that takes User Registration Events and creates an item in your database using Prisma.
+
+**Here are the steps:**
+
+- Navigate to `mqtt-consumer/src/index.js` and uncomment Lines 18 and 41-49, delete or comment out Line 52.
```

Comment:
let's replace this with explicit uncomments just incase line numbers change. So perhaps something along the lines of 
```
1. uncomment `import { PrismaClient } from "@prisma/client";` ,
2. uncomment everything after `//initialize and connect prisma client` 
3. comment out `let prismaClient = null;`